lib/gs-plugin-loader.c

```diff
@@ -1727,6 +1765,10 @@ gs_plugin_loader_pending_apps_remove (GsPluginLoader *plugin_loader,
 		}
 
 	}
+	if (save_queue) {
+		g_clear_pointer (&locker, g_mutex_locker_free);
+		save_install_queue (plugin_loader);
+	}
 	g_idle_add (emit_pending_apps_idle, g_object_ref (plugin_loader));
```

Comment:
Is it okay that this line may be within or outside the mutex depending on the state of save_queue?

Reply:
And, just to be sure, can you answer this question, too.

Reply:
As far as I know, this just adds the callback + args to be called with idle priority, so it should return before the function is called, which would unlock the mutex. There are other places where this is called while locked, but I can look into how glib manages all this if needed (not today though).